src/Symfony/Bridge/Doctrine/Tests/ContainerAwareEventManagerTest.php

```diff
@@ -40,29 +40,21 @@ public function testDispatchEventRespectOrder()
         $this->assertSame([$listener1, $listener2], array_values($this->evm->getListeners('foo')));
     }
 
-    /**
-     * @group legacy
-     */
-    public function testDispatchEventRespectOrderWithSubscribers()
+    public function testUsingDoctrineSubscribersThrows()
     {
-        $this->evm = new ContainerAwareEventManager($this->container, ['sub1', 'sub2']);
+        $this->evm = new ContainerAwareEventManager($this->container, [new MySubscriber(['foo'])]);
 
-        $this->container->set('sub1', $subscriber1 = new MySubscriber(['foo']));
-        $this->container->set('sub2', $subscriber2 = new MySubscriber(['foo']));
-
-        $this->expectDeprecation('Since symfony/doctrine-bridge 6.3: Using Doctrine subscribers as services is deprecated, declare listeners instead');
-        $this->assertSame([$subscriber1,  $subscriber2], array_values($this->evm->getListeners('foo')));
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage('Using Doctrine subscriber "Symfony\Bridge\Doctrine\Tests\MySubscriber" is not allowed, declare it as a listener instead.');
+        $this->evm->getListeners('foo');
     }
 
     public function testDispatchEvent()
     {
-        $this->container->set('lazy1', $listener1 = new MyListener());
```

Comment:
Why are you changing all those tests ? When removing deprecated features, non-legacy tests should not change at all.

Reply:
`EventManager::addEventListener()` now only accepts an object as the second argument. Because of this, there's no need to set services in the container and we must pass the listener object directly to the method

Reply:
But this would mean we would have to deprecate passing a string in 6.4 (and then, is this still lazy-loading the listeners, which is what this whole class is about ?). We cannot remove a non-deprecated API.

Reply:
Side note: it is still valid to widen a parameter type in a child class. So we are not forced to use `object` as type there. We could use `object|string` if we need it for our lazy-loading API.

Reply:
Alright, I missed the lazy-loading part. Thank you, I updated and reverted accordingly :+1:

Reply:
That's exactly why it is very important to not do any change in non-legacy tests in this kind of PR removing deprecated code, so that the testsuite keeps telling you about the code that is not allowed to regress.